pkg/controller/podgc/gc_controller_test.go

```diff
@@ -568,24 +607,74 @@ func TestGCTerminating(t *testing.T) {
 					},
 				})
 			}
-
+			pods := make([]*v1.Pod, 0, len(test.pods))
 			for _, pod := range test.pods {
 				creationTime = creationTime.Add(1 * time.Hour)
-				podInformer.Informer().GetStore().Add(&v1.Pod{
+				pods = append(pods, &v1.Pod{
 					ObjectMeta: metav1.ObjectMeta{Name: pod.name, CreationTimestamp: metav1.Time{Time: creationTime},
 						DeletionTimestamp: pod.deletionTimeStamp},
 					Status: v1.PodStatus{Phase: pod.phase},
 					Spec:   v1.PodSpec{NodeName: pod.nodeName},
 				})
 			}
+			client := setupNewSimpleClient(nodes, pods)
+			gcc, podInformer, nodeInformer := NewFromClient(client, -1)
 
-			gcc.gc(context.TODO())
-			deletedPodNames := getDeletedPodNames(client)
-
-			if pass := compareStringSetToList(test.deletedPodNames, deletedPodNames); !pass {
-				t.Errorf("[%v]pod's deleted expected and actual did not match.\n\texpected: %v\n\tactual: %v",
-					i, test.deletedPodNames.List(), deletedPodNames)
+			for _, pod := range pods {
+				podInformer.Informer().GetStore().Add(pod)
 			}
+			for _, node := range nodes {
+				nodeInformer.Informer().GetStore().Add(node)
+			}
+
+			gcc.gc(context.TODO())
```

Comment:
this may leak goroutines, can we create a context with cancel and defer the cancel()?

Reply:
indeed, however, this is a pre-existing issue, moved line from here: https://github.com/kubernetes/kubernetes/blob/9f9602ef830d55e844ad500c0a3ab342dd7cc98d/pkg/controller/podgc/gc_controller_test.go#L582

I will create a dedicated issue / pr to cleanup the test. Similar to this one: https://github.com/kubernetes/kubernetes/issues/111308

Reply:
yeah, it seems you hit a few of those :upside_down_face: